### ISSUE TITLE
Need a way to do command-click

### ISSUE THREAD

**Issue by blackketter** [dexterleng/vimac#204]
The command key is used to modify the click to be a double-click in Vimac.  I'd love for there to be a way to do a command-click.  Maybe the option key could be used.

(This also argues for the need to customize the trigger and modifier keys in the preferences.  In my daily usage, command-click is much more common than double-click.  I'd map command to command click and option to double-click.)

**Comment by dexterleng** [comment]
Customizing modifier keys sounds doable.

Another solution, instead of attempting to "overwrite" what a modifier does, Vimac should let them be dispatched normally, and double-click can be triggered through other means.

e.g.

right click can be done through cmd+typing the hints, but Vimac can allow shift+typing the hint to right click
double click can be done through double tapping the last hint letter.

**Comment by dexterleng** [comment]
@blackketter are you aware you can SHIFT-<type the hint letters> to right click? It should be functionally the same as right clicking, right?

**Comment by blackketter** [comment]
Right, but I'm talking about command-click not control-click, which is the same as right click.  Command-click is used for opening a link in a new tab in Safari, for example.

I like the idea of not overriding the modifiers and double-tapping the last hint letter, assuming that the hints are constructed so that there are no double letters at the end of a hint by default to avoid ambiguity.

**Comment by dexterleng** [comment]
ah ok.

I think the settings could look like:

Overwrite modifiers:

CTRL:
- [x] fall-through
- [ ] double-click
- [ ] right-click

CMD
- [x] fall-through
- [ ] double-click
- [ ] right-click

SHIFT
- [x] fall-through
- [ ] double-click
- [ ] right-click

**Comment by blackketter** [comment]
That would definitely work, but seems complicated.  (The fewer preferences the better. :) 

I don't have a sense for how the hints are constructed, but maybe a prefix to the hint would set up a double-click.  Let's say you reserved D to mean double click.  So if a hint was PR, pressing DPR would do a double click for PR.  You could also reserve other letters to set up the other modifiers.   So if an item that was hinted PR, typing:

- PR would click on that target
- DPR would do a double click on that target
- CPR would do a control click on that target (I believe that this is always the same as right-click, but I may be wrong)
- OPR would do an option click on that target
- SPR would do a shift click on that target
- MPR would do a command click on that target

And these could be additive, so DCPR could do a control-double-click.

Is this crazy?  Or even more complicated?

**Comment by blackketter** [comment]
And seeing #217, another prefix letter could enter scroll mode with the mouse on a target.  So "LPR" would move the mouse to that hint and start a scroll.

**Comment by dexterleng** [comment]
@blackketter do you have any examples of where you find yourself using `Command+Click` or other `Modifier+Click` with the mouse/trackpad?

**Comment by blackketter** [comment]
Like shift-click to extend a selection, command-click is for discontinuous selection of items like  Finder icons, Mail messages, OmniFocus items, spreadsheet cells, etc... 

Use it all the time.

**Comment by eret9616** [comment]
need command + click ..
It very often use in web browsing, like open in new tab

**Comment by eret9616** [comment]
j'

**Comment by dexterleng** [comment]
This is a good point. I was thinking about optimizing the workflow to click on the first item of menu which is "Open in new tab" for links, but I totally forgot you can just command+click.

**Comment by eret9616** [comment]
```
| Action      | How to trigger |
|-----------|-------------|
| Left click | Type the assigned hint-text |
| Right click | Type the assigned hint-text while holding `Shift` |
| Double left click | Type the assigned hint-text while holding `Command` |
| Move cursor | Type the assigned hint-text while holding `Option` |
| Rotate hints | `Space` |
| Exit | `Escape` or `Control + [`|
```

so .. ```command``` key is used to perform  double-click now, I think maybe..  use ```command``` to command-click... and ```control``` to double-click

**Comment by dexterleng** [comment]
@eret9616  there's a discussion in #331 on how we can support command+click. It will involve revamping the workflow.
